skorch/net.py

```diff
@@ -469,6 +495,16 @@ def initialize_optimizer(self):
 
         self.optimizer_ = self.optimizer(*args, **kwargs)
 
+        self.register_runtime_param(
+            ['optimizer__lr', 'lr'],
+            partial(self._set_lr, optimizer_attr='optimizer_', param_group=0)
+        )
+
+        self.register_runtime_param(
+            'optimizer__param_groups__(?P<param_group>[0-9]*)__lr',
```

Comment:
This introduces a new interface for targeting parameters in a parameter group. Maybe this should be tested, with a simple case such as: 

```python
optimizer__param_groups=[
        ('rnn*.period', {'lr': 0.3, 'momentum': 0}),
        ('rnn0', {'lr': 0.1})
]
```

And then setting `optimizer__param_groups__0__lr`.

Currently, this only coverts `lr` and other parameters such as `momentum`. The regex can most likely be expanded to cover any keyword, such as `momentum`.